python/google/protobuf/internal/well_known_types_test.py

```diff
@@ -372,6 +372,54 @@ def testInvalidDuration(self):
         r'Duration is not valid\: Sign mismatch.',
         message.ToJsonString)
 
+  def testTimestampOrdering(self):
+    a, b = timestamp_pb2.Timestamp(), timestamp_pb2.Timestamp()
+    # normalized + non-normalized comparisons
+    for a.seconds, a.nanos, b.seconds, b.nanos in ((0, 1, 1, 0), (1, 0, 0, 1000000001)):
+        self.assertTrue(a == a)
```

Comment:
These assertions could be factored out into a method that takes (a, b), then the two test methods can be DRY and take little extra work to add (datetime, Timestamp) comparison tests.

It could also have `a == deepcopy(a)`.